specs/vep-1722-properties-based-vdk-configuration-provider/README.md

```diff
@@ -0,0 +1,149 @@
+One issue is the lack of flexibility, as users are unable to adopt a new database plugin unless it has been pre-configured by a system administrator.
+This creates an additional step for users, requiring them to request the necessary configuration changes from administrators.
+In addition, the current method of setting configuration values via the Control Service introduces security concerns,
+as it requires the sharing of sensitive information such as passwords, and can lead to a poor user experience.
+
+
+
+
+## Requirements and goals
+<!--
+It tells **what** is it trying to achieve?
+List the specific goals (functional and nonfunctional requirements)? How will we
+know that this has succeeded?
+
+Specify non-goals. Clearly, the list of non-goals can't be exhaustive.
+Non-goals are only features, which a contributor can reasonably assume were a goal.
+One example is features that were cut during scoping.
+-->
+The new provider for VDK configuration should satisfy the following requirements and goals:
+
+* Allow users to set job-specific configuration using properties with the "vdk.config." prefix.
+* Have lesser priority than environment variables and the configuration values from  config.ini file.
+* Be flexible and secure in setting configuration, especially for sensitive data.
+
+
+### Non-goal
+
+In the future, config.ini file can be auto-generated entirely from the properties and updated when config.ini is edited on deploy.
+
+
+
+## High-level design
+
+The new provider for VDK configuration will be added as a new option in the VDK configuration system.
+It will take configuration from properties with the "vdk.config." prefix and have lesser priority than environment variables and the config.ini file.
+The properties will be stored in a properties file.
+
+[![vdk properties based config plugin](diagrams/vdk-properties-based-config-plugin.svg)](https://www.plantuml.com/plantuml/uml/TP31IiD048Rl-nG_yQA7z04yIAK5qNee53qLR9fCqzLaPjcThOBuxkvQZ3NYSVz_v_k6cGQjlCQ-CoeEAjYAglHehATQhR0nTIUnGJ4JLYyTLkGFjgO2xmO0QkcTCB6YM7fnvDLImFNojZ0VnfoqZ_C5PlSfpk2AsuwEsRpI2dUoBSQVL_F5j-HavD2yb6uGbRNmphPhTbriBHUe0dAx_7hYxCOJSLvaqACih_wiYai4KYGsVjXujRhFsGH7ooILjBDlviJm3sr7Mk_7lFFC-eswM7CJlUKMcrG-7SleQGEthmAPuH1PDBfNqZm4yl2H0uR91i-oDLDAtBxx1000)
+
+## API design
+
+The new provider for VDK configuration will not require any changes to existing API.
+
+
+
+## Detailed design
+
+
+[![sequence diagram](diagrams/sequence-diagram.svg)](https://www.plantuml.com/plantuml/uml/ZLCzZ-8m4EtzAqPGKC59b1Gd3X3IVHGqTnNIoOadYPTWM_OO_lvEecZZJKYqI4XuFjwRzsRO1vAUuhqMCYVhuMz0Bno7TAwTD0IBVwVVSFppS_4vUl7M8Rynm8yB9qSjIS7HcaANSAbZgSsujCj7BqdRL_cJ90c_RBP8PWGP86w5Q7-ll_TZmWuUwWQkfxyQMAy34dzpZmGhpc_obdD3NXlxtyaGdjQhLKEW7v8mmLkUpmV-Y3x2T4Fzp1hGg02Js6BKCW732d46X-Le0ohxbidynfHjWonHA1mFB-AMQGOkTWxvo31DxCrclUv1wV9mQEIBqjleeCdbtR8GN2K3YjbbJuaiaFAAdxAkqqsfo5BB4SLBv-PfT-2HeZT3Yg-CEpv9XekElSBGMP0auI7hYE5g2kk7Ujh_LgFegQbALohGgCHJ7qtgPc__z3XXTXQoBIPLs6wtOHiZystiFYwWhI-qqQ52DRobla_i6IlUwtS0)
+
```

Comment:
nit: users actors are all over the place; the flow is sequential, or could be -async